bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorSerialConnector.java

```diff
@@ -0,0 +1,220 @@
+ */
+package org.openhab.binding.epsonprojector.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.TooManyListenersException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorException;
+import org.openhab.core.io.transport.serial.PortInUseException;
+import org.openhab.core.io.transport.serial.SerialPort;
+import org.openhab.core.io.transport.serial.SerialPortEvent;
+import org.openhab.core.io.transport.serial.SerialPortEventListener;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.io.transport.serial.UnsupportedCommOperationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector for serial port communication.
+ *
+ * @author Pauli Anttila - Initial contribution
+ * @author Michael Lobstein - Improvements for OH3
+ */
+@NonNullByDefault
+public class EpsonProjectorSerialConnector implements EpsonProjectorConnector, SerialPortEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorSerialConnector.class);
+    private final String serialPortName;
+    private final SerialPortManager serialPortManager;
+
+    private @Nullable InputStream in = null;
+    private @Nullable OutputStream out = null;
+    private @Nullable SerialPort serialPort = null;
+
+    public EpsonProjectorSerialConnector(SerialPortManager serialPortManager, String serialPort) {
+        this.serialPortManager = serialPortManager;
+        this.serialPortName = serialPort;
+    }
+
+    @Override
+    public void connect() throws EpsonProjectorException {
+        try {
+            logger.debug("Open connection to serial port '{}'", serialPortName);
+
+            SerialPortIdentifier serialPortIdentifier = serialPortManager.getIdentifier(serialPortName);
+
+            if (serialPortIdentifier == null) {
+                throw new IOException("Unknown serial port");
+            }
+            SerialPort serialPort = serialPortIdentifier.open(this.getClass().getName(), 2000);
+            serialPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            serialPort.enableReceiveThreshold(1);
+            serialPort.disableReceiveTimeout();
+
+            InputStream in = serialPort.getInputStream();
+            OutputStream out = serialPort.getOutputStream();
+
+            if (in != null && out != null) {
+                out.flush();
+                if (in.markSupported()) {
+                    in.reset();
+                }
+
+                // RXTX serial port library causes high CPU load
+                // Start event listener, which will just sleep and slow down event loop
+                serialPort.addEventListener(this);
```

Comment:
This bug has been fixed. So, the workaround could be removed.

Reply:
Done, but I checked and it is still in about 8 other bindings.

Reply:
It doesn't seem to hurt, but it's not necessary.